crates/fluvio-protocol-derive/src/util.rs

```diff
@@ -1,75 +1,273 @@
+                    span,
+                    format!("Expected {attr_name} attribute to be a LitStr or LitInt: `{attr_name} = \"...\"`"),
+                ))
             }
         }
+        Some(Expr::Unary(ExprUnary { expr, .. })) => {
+            // When passing -1 as a value it is returned as type Unary
+            // So to handle that we are checking if it's Unary Lit and continue as usual
+            // If needed this can be extended to handle the Unary operators
+            // But it doesn't seem that is necessary currently
+            Ok(PropAttrsType::Int(-parse_int_expr(attr_name, expr, span)?))
+        }
+        Some(Expr::Path(ExprPath { path, .. })) => {
+            // For now we only need Path just to handle cases where CONSTANTS are being passed without "" quotes
+            if let Some(path_ident) = path.get_ident() {
+                return Ok(PropAttrsType::Lit(syn::Ident::new(
+                    &path_ident.to_string(),
+                    path_ident.span(),
+                )));
+            }
+
+            Err(syn::Error::new(
+                span,
+                format!("Attribute {attr_name} needs to be a valid Path: `{attr_name} = \"...\"`"),
+            ))
+        }
+        _ => Err(syn::Error::new(
+            span,
+            format!(
+                "Expected {attr_name} attribute to be a Lit/Path/Unary: `{attr_name} = \"...\"`"
+            ),
+        )),
     }
+}
 
-    None
+fn parse_int_expr<'a, T>(attr_name: &'a str, expr: &'a Expr, span: Span) -> syn::Result<T>
+where
+    T: std::ops::Neg<Output = T> + FromStr,
+    <T as FromStr>::Err: std::fmt::Display,
+{
+    match expr {
+        Expr::Unary(ExprUnary {
+            op: UnOp::Neg(_),
+            expr,
+            ..
+        }) => parse_int_expr::<T>(attr_name, expr, span).map(|int| -int),
+        Expr::Lit(ExprLit {
+            lit: Lit::Int(int), ..
+        }) => int.base10_parse(),
+        _ => Err(syn::Error::new(
+            span,
+            format!("Expected {attr_name} to be Int: `{attr_name} = \"...\"`"),
+        )),
+    }
 }
 
-/// find name value with integer value
-pub(crate) fn find_int_name_value(version_meta: &Meta, attr_name: &str) -> Option<u64> {
-    if let Some(attr) = find_name_attribute(version_meta, attr_name) {
-        match &attr.lit {
-            Lit::Int(version_val) => {
-                //  println!("version value: {}",version_val.value());
-                version_val.base10_parse::<u64>().ok()
+pub fn get_expr_int_value<'a, T>(
+    attr_name: &'a str,
+    value: Option<&'a Expr>,
+    span: Span,
+) -> syn::Result<T>
+where
+    T: std::ops::Neg<Output = T> + FromStr,
+    <T as FromStr>::Err: std::fmt::Display,
+{
+    if let Some(value_expr) = value {
+        parse_int_expr::<T>(attr_name, value_expr, span)
+    } else {
+        Err(syn::Error::new(
+            span,
+            format!("Expected {attr_name} to be valid Expr: `{attr_name} = \"...\"`"),
+        ))
+    }
```

Comment:
```suggestion
    let value_expr = value.ok_or_else(|| {
        syn::Error::new(
            span,
            format!("Expected {attr_name} to be valid Expr: `{attr_name} = \"...\"`"),
        )
    })?;
    parse_int_expr::<T>(attr_name, value_expr, span)
```

Reply:
Removed the function because we don't need it anymore, because we have no more `Option<&Expr>` so no need for a separate function to unwrap, using directly the `parse_int_expr`.